testing/camunda-process-test-java/src/test/java/io/camunda/process/test/impl/extensions/CamundaProcessTestContextIT.java

```diff
@@ -0,0 +1,181 @@
+    CamundaAssert.assertThat(processInstanceEvent).hasCompletedElements("success-end");
+    CamundaAssert.assertThat(processInstanceEvent).hasVariables(variables);
+  }
+
+  @Test
+  void shouldMockJobWorkerWithJobHandlerBpmnError() {
+    // Given
+    processTestContext
+        .mockJobWorker("test")
+        .withHandler(
+            (jobClient, job) -> {
+              jobClient.newThrowErrorCommand(job).errorCode("bpmn-error").send().join();
+            });
+    final long processDefinitionKey = deployProcessModel();
+
+    // When
+    final ProcessInstanceEvent processInstanceEvent =
+        client.newCreateInstanceCommand().processDefinitionKey(processDefinitionKey).send().join();
+
+    // Then
+    CamundaAssert.assertThat(processInstanceEvent).isCompleted();
+    CamundaAssert.assertThat(processInstanceEvent).hasCompletedElements("error-end");
+  }
+
+  @Test
+  void shouldMockJobWorkerWithJobHandlerSuccess() {
+    // Given
+    processTestContext
+        .mockJobWorker("test")
+        .withHandler(
+            (jobClient, job) -> {
+              jobClient.newCompleteCommand(job).send().join();
+            });
+    final long processDefinitionKey = deployProcessModel();
+
+    // When
+    final ProcessInstanceEvent processInstanceEvent =
+        client.newCreateInstanceCommand().processDefinitionKey(processDefinitionKey).send().join();
+
+    // Then
+    CamundaAssert.assertThat(processInstanceEvent).isCompleted();
+    CamundaAssert.assertThat(processInstanceEvent).hasCompletedElements("success-end");
+  }
+
+  /**
+   * Deploys a process model and waits until it is accessible via the API.
+   *
+   * @return the process definition key
+   */
+  private long deployProcessModel() {
+    final BpmnModelInstance processModel =
+        Bpmn.createExecutableProcess("test-process")
+            .startEvent()
+            .serviceTask("service-task-1")
+            .zeebeJobType("test")
+            .boundaryEvent("error-boundary-event")
+            .error("bpmn-error")
+            .endEvent("error-end")
+            .moveToActivity("service-task-1")
+            .endEvent("success-end")
+            .done();
+
+    final DeploymentEvent deploymentEvent =
+        client
+            .newDeployResourceCommand()
+            .addProcessModel(processModel, "test-process.bpmn")
+            .send()
+            .join();
+    final long processDefinitionKey =
+        deploymentEvent.getProcesses().stream().findFirst().get().getProcessDefinitionKey();
+    Awaitility.await("until process model is accessible via API")
+        .atMost(Duration.ofSeconds(2 * TIMEOUT))
+        .untilAsserted(
+            () -> {
+              final Future<?> futureRequest =
+                  client.newProcessDefinitionGetRequest(processDefinitionKey).send();
+              assertThat(futureRequest).succeedsWithin(Duration.ofSeconds(TIMEOUT));
+            });
+    return processDefinitionKey;
```

Comment:
:question: Why do we need to wait until the process is available?

In other test cases, we don't wait and create the process instances using the BPMN process id.